File: tapnpay_flutter_example/lib/models/fawry_transaction_request.dart
/// Transaction input collected by the demo UI, mirroring
/// `FawryTransactionRequest` in `example/sdk-integration`.
class FawryTransactionRequest {
  const FawryTransactionRequest({
    required this.amount,
    required this.btc,
    required this.orderId,
    required this.merchantToken,
    required this.merchantAccountNumber,
    this.referenceNumber = '',
  });

  final String amount;
  final String btc;
  final String orderId;
  final String merchantToken;
  final String merchantAccountNumber;
  final String referenceNumber;
}

File: example/sdk-integration/src/main/java/com/fawry/softpos/android/example/sdk/FawryTransactionRequest.kt
package com.fawry.softpos.android.example.sdk

/**
 * Transaction input collected by the demo UI before it is mapped to Fawry SDK builders.
 *
 * Values are strings to keep this sample close to form input. Client apps can validate and convert
 * them earlier if their production flow has stricter requirements.
 */
data class FawryTransactionRequest(
    val amount: String,
    val btc: String,
    val orderId: String,
    val merchantToken: String,
    val merchantAccountNumber: String,
    val referenceNumber: String,
)
